godot-core/src/builtin/rect2i.rs

```diff
@@ -78,21 +82,492 @@ impl Rect2i {
+        self.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+        let mut end = self.end();
+        let mut begin = self.position;
+
+        begin.x = cmp::min(begin.x, to.x);
+        begin.y = cmp::min(begin.y, to.y);
+        end.x = cmp::max(end.x, to.x);
+        end.y = cmp::max(end.y, to.y);
+
+        Self::from_corners(begin, end)
+    }
+
+    /// Returns the area of the `Rect2i`.
+    #[inline]
+    pub const fn get_area(&self) -> i32 {
+        self.size.x * self.size.y
+    }
+
+    /// Returns the center of the `Rect2i`, which is equal to `position + (size / 2)`.
+    ///
+    /// If `size` is an odd number, the returned center value will be rounded towards `position`.
+    #[inline]
+    pub fn get_center(&self) -> Vector2i {
+        self.position + (self.size / 2)
+    }
+
+    /// Returns a copy of the `Rect2i` grown by the specified `amount` on all sides.
+    #[inline]
+    pub fn grow(self, amount: i32) -> Self {
+        let amount_2d = Vector2i::new(amount, amount);
+        Self::from_corners(self.position - amount_2d, self.end() + amount_2d)
+    }
+
+    /// Returns a copy of the `Rect2i` grown by the specified amount on each side individually.
+    #[inline]
+    pub const fn grow_individual(self, left: i32, top: i32, right: i32, bottom: i32) -> Self {
+        Self::from_components(
+            self.position.x - left,
+            self.position.y - top,
+            self.size.x + left + right,
+            self.size.y + top + bottom,
+        )
+    }
+
+    /// Returns a copy of the `Rect2i` grown by the specified `amount` on the specified `RectSide`.
+    #[inline]
+    pub const fn grow_side(self, side: RectSide, amount: i32) -> Self {
+        match side {
+            RectSide::Left => self.grow_individual(amount, 0, 0, 0),
+            RectSide::Top => self.grow_individual(0, amount, 0, 0),
+            RectSide::Right => self.grow_individual(0, 0, amount, 0),
+            RectSide::Bottom => self.grow_individual(0, 0, 0, amount),
+        }
+    }
+
+    /// Returns `true` if the `Rect2i` has area, and `false` if the `Rect2i` is linear, empty, or
+    /// has a negative `size`.
+    #[inline]
+    pub const fn has_area(&self) -> bool {
+        self.size.x > 0 && self.size.y > 0
+    }
+
+    /// Returns `true` if the `Rect2i` contains a point. By convention, the right and bottom edges
+    /// of the `Rect2i` are considered exclusive, so points on these edges are not included.
+    #[inline]
+    pub const fn has_point(&self, point: Vector2i) -> bool {
+        self.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+
+        let own_end = self.end();
+        point.x >= self.position.x
+            && point.y >= self.position.y
+            && point.x < own_end.x
+            && point.y < own_end.y
+    }
```

Comment:
Please rename to `contains_point`.

Also, there's no ambiguity as to which end is meant, so `own_end` could simply be `end` (points have no beginnings/ends).

Reply:
Given that the relevant Gdscript function is `has_point`, is there a particular reason for renaming it?

Reply:
Consistency with `Array::contains()`, `PackedArray::contains()`, `Dictionary::contains_key()` -- they are also `has*` in Godot, but `contains*` for Rust collections.

Also, "has" is a really bad word because it means so many things. For example, this very class offers `has_area()` and `has_point()`, where it means two entirely different things ("fulfills a certain property" and "includes/contains another geometry").

What we can do is add [`#[doc(alias = "has_point")]`](https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#alias) to allow finding it by its Godot name. Furthermore, we typically add a line to elaborate the GDScript equivalent, see here:
https://github.com/godot-rust/gdext/blob/e3b9ad1433eed0d83c144cbead2c3d014ea6d0d4/godot-core/src/builtin/dictionary.rs#L123-L126